src/TogglePassword/src/Form/TogglePasswordTypeExtension.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\TogglePassword\Form;
+
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Translation\TranslatableMessage;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * @author Félix Eymonot <felix.eymonot@alximy.io>
+ */
+final class TogglePasswordTypeExtension extends AbstractTypeExtension
+{
+    public function __construct(private readonly ?TranslatorInterface $translator)
+    {
+    }
+
+    public static function getExtendedTypes(): iterable
+    {
+        return [PasswordType::class];
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'toggle' => false,
+            'hidden_label' => 'Hide',
+            'visible_label' => 'Show',
+            'hidden_icon' => 'Default',
+            'visible_icon' => 'Default',
+        ]);
+        $resolver->setAllowedTypes('toggle', ['boolean']);
+        $resolver->setAllowedTypes('hidden_label', ['string', TranslatableMessage::class, 'null']);
+        $resolver->setAllowedTypes('visible_label', ['string', TranslatableMessage::class, 'null']);
+        $resolver->setAllowedTypes('hidden_icon', ['string', 'null']);
+        $resolver->setAllowedTypes('visible_icon', ['string', 'null']);
+    }
+
+    public function finishView(FormView $view, FormInterface $form, array $options): void
+    {
+        if (!$options['toggle']) {
+            return;
+        }
+
+        $attr = $view->vars['attr'] ?? [];
+        $controllerName = 'symfony--ux-toggle-password--toggle-password';
+        $attr['data-controller'] = trim(sprintf('%s %s', $attr['data-controller'] ?? '', $controllerName));
+
+        $controllerValues = [];
+        $controllerValues['hidden-label'] = $this->translateLabel($options['hidden_label']);
+        $controllerValues['visible-label'] = $this->translateLabel($options['visible_label']);
+        $controllerValues['hidden-icon'] = $options['hidden_icon'];
+        $controllerValues['visible-icon'] = $options['visible_icon'];
+        if (isset($attr['toggle_classes'])) {
+            $controllerValues['button-classes'] = json_encode(
+                explode(' ', $attr['toggle_classes']),
+                \JSON_THROW_ON_ERROR
+            );
+            unset($attr['toggle_classes']);
+        }
```

Comment:
I think it would be better to have a proper option (e.g. `button_classes` requiring an array of strings) instead of looking for a specific attr.

Reply:
Ok, actually i was hesitating between this approach (closer to classic Form usage) and your suggestion 😄 so let's change that !

Reply:
Well, it does not really apply to the input in the end :)